PROCESSORS.md

```diff
@@ -1125,3 +1126,40 @@ In the list below, the names of required properties appear in bold. Any other pr
 |success|All files are routed to success|
 
 
+## PutS3Object
+
+### Description
+
+Puts FlowFiles to an Amazon S3 Bucket. The upload uses either the PutS3Object method. The PutS3Object method send the file in a single synchronous call, but it has a 5GB size limit. Larger files sent using the multipart upload methods are currently not supported. The AWS libraries select an endpoint URL based on the AWS region, but this can be overridden with the 'Endpoint Override URL' property for use with other S3-compatible endpoints. The S3 API specifies that the maximum file size for a PutS3Object upload is 5GB.
+### Properties
+
+In the list below, the names of required properties appear in bold. Any other properties (not in bold) are considered optional. The table also indicates any default values, and whether a property supports the NiFi Expression Language.
+
+| Name | Default Value | Allowable Values | Description |
+| - | - | - | - |
+|**Object Key**|${filename}||The key of the S3 object<br/>**Supports Expression Language: true**|
```

Comment:
Sounds good to me, I am not familiar how common are agents without EL support, but it's reasonable to support those as well. In that case the access control properties should lose the default values too, as if those are invalid the put operation fails on AWS. Those properties are not mandatory so we can have empty defaults if that sounds okay?

Reply:
I'm not familiar enough with S3 to know this, but if empty is a reasonable default that's usable in simple setups, then I support it.

Reply:
These values control the permissions of the uploaded object. We can give email addresses or AWS IDs that should have read/write/full control permissions on the uploaded object, to which by default only the owner (the uploader's account) have access if no other permissions are given. I think that should be okay in our case. The current default values like ${s3.permissions.full.users} are from the Nifi version of PutS3Object which can probably come from other AWS processors, but we don't have those yet, so we can remove those.

Reply:
Interesting point. NiFi compatibility would be nice, but in NiFi, expression language is not an optional feature. That made me neutral, I'm fine with either option.

Reply:
I searched for the `s3.permission` parameters in NiFi if any of the processors have them as outputs, but I could not find any, only as the default input values of PutS3Object. I am not sure, where these defaults come from, but I suppose in this case it would be more advantageous to have support for agents without EL support if we cannot find any other references to them. I removed the EL dependent default values in [bc6b752](https://github.com/apache/nifi-minifi-cpp/pull/915/commits/bc6b752e03ddd29dc150cbb598414cb332ca5592)